admiral/pkg/apis/admiral/routes/handlers.go

```diff
@@ -27,8 +28,32 @@ type IdentityServiceEntry struct {
 	ClusterNames []string `json:"Clusters,omitempty"`
 }
 
+
+/*
+We expect the DNS health checker to include the query param checkifreadonly with value set to true.
+The query param is used to check if the current Admiral instance is running in Active Mode or Passive Mode (also called read only mode).
+If Running in passive  mode, the health check returns 502 which forces DNS lookup to always return reference to Admiral in Active state.
+*/
+
 func (opts *RouteOpts) ReturnSuccessGET(w http.ResponseWriter, r *http.Request) {
-	w.WriteHeader(200)
+	allQueryParams:= r.URL.Query()
+	checkIfReadOnlyStringVal := allQueryParams.Get("checkifreadonly")
+	//Remove all spaces
+	checkIfReadOnlyStringVal = strings.ReplaceAll(checkIfReadOnlyStringVal," ","")
+	// checkIfReadOnlyStringVal will be empty in case ""checkifreadonly" query param is not sent in the request. checkIfReadOnlyBoolVal will be false
+	checkIfReadOnlyBoolVal, _ := strconv.ParseBool(checkIfReadOnlyStringVal)
+	if checkIfReadOnlyBoolVal{
```

Comment:
@asushanthk , I'm sorry maybe I'm missing something but why do we need this query param?
since the active/passive state is stored in `AdmiralState`, the correct instance of the admiral should be able to return its current state.
I'm wondering if the healthcheck URLs are not set up correctly, i.e., this query param is missed, in that case both the admiral instances will end up returning 200.

Reply:
The same health check request will be used for the readiness and liveness probe. The health check API should return 200 to keep the passive pods running.
DNS checker can send the extra query param to support pinning traffic to the active pod